maps/southern_cross/events/wildlife_encounter.dm

```diff
@@ -0,0 +1,95 @@
+// This event sends random sif mobs at someone in the wilderness, unannounced.
+
+/datum/event2/meta/wildlife_encounter
+	name = "random encounter"
+	departments = list(DEPARTMENT_EVERYONE)
+	chaos = 10
+	event_type = /datum/event2/event/wildlife_encounter
+
+/datum/event2/meta/wildlife_encounter/get_weight()
+	var/explorers = metric.count_people_with_job(/datum/job/explorer) + metric.count_people_with_job(/datum/job/sar)
+
+	if(!explorers)
+		return 0
+	return 20 + explorers * 50
+
+/datum/event2/event/wildlife_encounter
+	var/mob/living/victim = null
+
+/datum/event2/event/wildlife_encounter/set_up()
+	var/list/potential_victims = list()
+
+	for(var/mob/living/L in player_list)
+		//if(!(L.z in get_location_z_levels()))
+		//	log_debug("Not on the right z-level")
+		//	continue // Not on the right z-level.
+
+		if(L.stat)
+			continue // Don't want dead people.
+		if(istype(get_turf(L), /turf/simulated/floor/outdoors) && istype(get_area(L),/area/surface/outside/wilderness))
+			potential_victims += L
+
+	if(potential_victims.len)
+		victim = pick(potential_victims)
+		potential_victims = null
+
+/datum/event2/event/wildlife_encounter/start()
+	if(!victim)
+		log_debug("Failed to find a target for random encounter. Aborting.")
+		abort()
+		return
+
+	var/number_of_packs = rand(1, 3)
+	// Getting off screen tiles is kind of tricky due to potential edge cases that could arise.
+	// The method we're gonna do is make a big square around the victim, then
+	// subtract a smaller square in the middle for the default vision range.
+
+	var/list/outer_square = get_safe_square(victim, world.view + 5)
+	var/list/inner_square = get_safe_square(victim, world.view + 1)
+
+	var/list/donut = outer_square - inner_square
+	for(var/T in donut)
+		if(!istype(T, /turf/simulated/floor/outdoors))
+			donut -= T
+
+	var/build_path = item_to_spawn()
+	var/turf/spawning_turf = pick(donut)
+	log_debug("Sending [number_of_packs] [build_path]\s after \the [victim].")
+	for(var/i = 1 to number_of_packs)
+
+		if(spawning_turf)
+			var/mob/living/simple_mob/M = new build_path(spawning_turf)
+			M.ai_holder?.give_destination(get_turf(victim))
```

Comment:
What happens if there's a suitable target closer to the mob than the selected victim (general AI pathing question)? Will a hungry spider walk _away_ from a potential meal in search of a more distant one? What if the closer meal starts shooting at the spider? (Last question is probably answered by "it'll turn around and bite back", but I would like you to verify that)

Reply:
give.destination is the same thing that the move command with build mode does. So the spider will stop moving and attack as soon as it sees a target.

Someone shooting it from off-screen won't cause it to change movement though, but if you're on screen it'll try and eat you even if you don't shoot it.